### PR TITLE
Refactoring api to use translations (chunk 1)

### PR DESCRIPTION
**Pull request by enahum**
- Add spanish translations
- Files in order by name from admin to export
- Does not include api templates and tests
- Fix web_test to load translations
- Fix i18n to fallback to DEFAULT_LOCALE if no system locale found

### CODE REVIEW

File: config/config.json [hunk 1]
```diff
@@ -107,5 +107,28 @@
...
+        "BindPassword": null,
+        "FirstNameAttribute": null,
+        "LastNameAttribute": null,
+        "EmailAttribute": null,
+        "UsernameAttribute": null,
+        "IdAttribute": null,
+        "QueryTimeout": 60
     }
-}
+}
```
Comment (jwilander) [inline config/config.json]: There shouldn't be any changes in config.json, this stuff is pulled over from the enterprise edition. Can you remove the GoogleSettings and LdapSettings?

**Comment by jwilander** [comment]
+1, other than my one comment

File: utils/i18n.go [hunk 2]
```diff
@@ -36,7 +36,10 @@ func InitTranslations() {
 func GetTranslationsBySystemLocale() i18n.TranslateFunc {
 	locale := model.DEFAULT_LOCALE
 	if userLanguage, err := jibber_jabber.DetectLanguage(); err == nil {
-		locale = userLanguage
+		// only set the system locale if is supported, fallback to DEFAULT_LOCALE
```
Comment (coreyhulen) [inline utils/i18n.go]: I don't think this is needed?  the next couple of lines should already be doing this.

```
if locales[locale] == "" {
		l4g.Error("Failed to load system translations for '%v' attempting to fall back to '%v'", locale, model.DEFAULT_LOCALE)

		if locales[model.DEFAULT_LOCALE] == "" {
			panic("Failed to load system translations for '" + model.DEFAULT_LOCALE + "'")
		}
	}
```

Comment (coreyhulen) [inline utils/i18n.go]: oops...wait there is a bug here.  It should be

```if locales[locale] == "" {
        l4g.Error("Failed to load system translations for '%v' attempting to fall back to '%v'", locale, model.DEFAULT_LOCALE)
        locale = model.DEFAULT_LOCALE
```

File: utils/i18n.go [hunk 3]
```diff
@@ -138,3 +141,12 @@ func getTranslationsAndLocale(w http.ResponseWriter, r *http.Request) (i18n.Tran
 	SetLocaleCookie(w, model.DEFAULT_LOCALE, 10)
 	return translations, model.DEFAULT_LOCALE
 }
+
+func contains(l string) bool {
```
Comment (coreyhulen) [inline utils/i18n.go]: contains is no longer needed.  I change `locales` to a map.

File: utils/i18n.go (hunk 2 above)
Comment (enahum) [inline utils/i18n.go]: It wasn't working I tried setting environment var LANG=fr and it didn't fallback to DEFAULT_LOCALE it just stated the the system translations failed to load, maybe if we change that to panic?

Comment (coreyhulen) [inline utils/i18n.go]: there was a bug,  I think we need to add the line `locale = model.DEFAULT_LOCALE` in the if block ?

**Comment by coreyhulen** [comment]
+1, we can merge once travis finishes.  I can fix the bug I added in `GetTranslationsBySystemLocale` with a separate PR.

**Comment by coreyhulen** [comment]
hah, nvm. looks like @enahum fixed it.

**Pull request merged**